src/Type/Constant/ConstantArrayType.php

```diff
@@ -693,6 +693,20 @@ public function getOffsetValueType(Type $offsetType): Type
 
 	public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $unionValues = true): Type
 	{
+		if ($offsetType !== null) {
+			$constantScalars = $offsetType->getConstantScalarTypes();
+			$constantScalarsCount = count($constantScalars);
+			if ($constantScalarsCount > 1 && $constantScalarsCount < ConstantArrayTypeBuilder::ARRAY_COUNT_LIMIT) {
+				$arrays = [];
+				foreach ($constantScalars as $constantScalar) {
+					$builder = ConstantArrayTypeBuilder::createFromConstantArray($this);
+					$builder->setOffsetValueType($constantScalar, $valueType);
+					$arrays[] = $builder->getArray();
+				}
+
+				return TypeCombinator::union($this, ...$arrays);
```

Comment:
I might be tired but why 
```
				return TypeCombinator::union($this, ...$arrays);
```
and not
```suggestion
				return TypeCombinator::union(...$arrays);
```
?

Reply:
I was also tired, it makes sense without the `$this` 😊 Updated that.